### PR TITLE
Handle App Service MSI response expires_on values

### LINKED ISSUE #5952: Normalize expires_on
`get_token` should always return `expires_on` as seconds since the epoch. IMDS, Cloud Shell, AAD (client secret and cert credentials) responses contain `expires_on` as seconds since the epoch or `expires_in`, and `get_token` does the right thing in those cases.

App Service MSI responses, however, contain only `expires_on` as a date/time string, e.g. `"06/19/2019 11:19:16 +00:00"`. These presently aren't converted to seconds since the epoch.

### PR DESCRIPTION
**Pull request by chlowell**
This closes #5952 with code to parse the expires_on strings returned in App Service MSI responses (e.g. "06/19/2019 23:42:01 +00:00") to epoch seconds, as required by `AccessToken`.

### CODE REVIEW

**Comment by adxsdk6** [comment]
Can one of the admins verify this patch?

File: sdk/identity/azure-identity/azure/identity/_authn_client.py [hunk 1]
```diff
@@ -62,15 +64,31 @@ def _deserialize_and_cache_token(self, response, scopes, request_time):
...
-
             # AccessToken contains the token's expires_on time. There are four cases for setting it:
             # 1. response has expires_on -> AccessToken uses it
             # 2. response has expires_on and expires_in -> AccessToken uses expires_on
             # 3. response has only expires_in -> AccessToken uses expires_in + time of request
-            # 4. response has neither expires_on or expires_in -> AccessToken sets expires_on = 0
+            # 4. response has neither expires_on nor expires_in -> AccessToken sets expires_on = 0
             #    (not expecting this case; if it occurs, the token is effectively single-use)
             expires_on = payload.get("expires_on", 0)
+            if not isinstance(expires_on, int):
```
Comment (xiangyan99) [inline sdk/identity/azure-identity/azure/identity/_authn_client.py]: Maybe we can remove this check.

Comment (chlowell) [reply sdk/identity/azure-identity/azure/identity/_authn_client.py]: Good idea. I simplified the code here.

**Pull request merged**